File: mobile/lib/shared/models/political_actor.dart
class PoliticalActor {
  final int id;
  final String source;
  final String sourceId;
  final String displayName;
  final String? party;
  final String? state;
  final String role;
  final String status;
  final String? photoUrl;
  final String? sourceUrl;
  final DateTime lastIndexedAt;

  const PoliticalActor({
    required this.id,
    required this.source,
    required this.sourceId,
    required this.displayName,
    required this.party,
    required this.state,
    required this.role,
    required this.status,
    required this.photoUrl,
    required this.sourceUrl,
    required this.lastIndexedAt,
  });

  factory PoliticalActor.fromJson(Map<String, dynamic> json) {
    return PoliticalActor(
      id: json['id'] as int,
      source: json['source'] as String,
      sourceId: json['source_id'] as String,
      displayName: json['display_name'] as String,
      party: json['party'] as String?,
      state: json['state'] as String?,
      role: json['role'] as String,
      status: json['status'] as String,
      photoUrl: json['photo_url'] as String?,
      sourceUrl: json['source_url'] as String?,
      lastIndexedAt: DateTime.parse(json['last_indexed_at'] as String),
    );
  }

  String get roleLabel {
    if (role == 'federal_deputy') return 'Deputada(o) federal';
    return role;
  }
}

class PoliticalActorListResponse {
  final List<PoliticalActor> actors;
  final int totalCount;
  final bool hasNext;

  const PoliticalActorListResponse({
    required this.actors,
    required this.totalCount,
    required this.hasNext,
  });

  factory PoliticalActorListResponse.fromJson(Map<String, dynamic> json) {
    return PoliticalActorListResponse(
      actors: (json['actors'] as List<dynamic>)
          .map((item) => PoliticalActor.fromJson(item as Map<String, dynamic>))
          .toList(),
      totalCount: json['total_count'] as int,
      hasNext: json['has_next'] as bool,
    );
  }
}

class TrendingPoliticalActor {
  final PoliticalActor actor;
  final int rank;

  const TrendingPoliticalActor({required this.actor, required this.rank});

  factory TrendingPoliticalActor.fromJson(Map<String, dynamic> json) {
    return TrendingPoliticalActor(
      actor: PoliticalActor.fromJson(json['actor'] as Map<String, dynamic>),
      rank: json['rank'] as int,
    );
  }
}

File: backend/app/core/entities/political_actor.py
from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class PoliticalActor:
    id: int
    source: str
    source_id: str
    normalized_name: str
    display_name: str
    party: str | None
    state: str | None
    role: str
    status: str
    photo_url: str | None
    source_url: str | None
    last_indexed_at: datetime


@dataclass(frozen=True)
class OfficialEvidence:
    id: int
    political_actor_id: int
    source: str
    source_id: str
    evidence_type: str
    title: str
    summary: str
    evidence_date: datetime | None
    source_url: str | None
    fetched_at: datetime
    expires_at: datetime


@dataclass(frozen=True)
class FollowedActor:
    anonymous_id: str
    political_actor_id: int
    updated_at: datetime


@dataclass(frozen=True)
class TrendingActor:
    actor: PoliticalActor
    rank: int
    follow_count: int


@dataclass(frozen=True)
class SectionSummary:
    summary: str
    citations: list[dict[str, str]]


@dataclass(frozen=True)
class SentinelaSummary:
    id: int
    political_actor_id: int
    period: str
    generated_at: datetime
    expires_at: datetime
    votes: SectionSummary
    propositions: SectionSummary
    expenses: SectionSummary
    synthesis: str
